zebra-chain/src/orchard/note/nullifiers.rs

```diff
@@ -40,7 +40,11 @@ pub struct Nullifier(#[serde(with = "serde_helpers::Base")] pallas::Base);
 
 impl From<[u8; 32]> for Nullifier {
     fn from(bytes: [u8; 32]) -> Self {
-        Self(pallas::Base::from_bytes(&bytes).unwrap())
+        // TODO:
+        // - previously this was just `unwrap()` but we started to get panics in prop tests
+        // - confirm the `unwrap_or()` is what we want here
+        // - change to the correct code and remove this TODO after confirmation
+        Self(pallas::Base::from_bytes(&bytes).unwrap_or(pallas::Base::zero()))
```

Comment:
@dconnolly thoughts on Pallas verification during deserialization?

Reply:
`from_u512` is a private method.

Reply:
> > * fix the `Arbitrary for Nullifier` impl to generate valid points using `from_u512`: https://github.com/zcash/pasta_curves/blob/main/src/fields/fp.rs#L252
> 
> `from_u512` is a private method.

It might be a lot faster to use `From<u64>` or `from_u128`.

We're not doing any operations on these points yet, and this isn't cryptographically secure code, so the bias avoided by u512/wide doesn't matter.

If it ever does matter, there's a public `from_bytes_wide` method, which calls `from_u512`:
https://github.com/zcash/pasta_curves/blob/main/src/fields/fp.rs#L709

@dconnolly any advice on the difference between converting from u64, u128, and u512?

Reply:
I made a fix for this at https://github.com/ZcashFoundation/zebra/pull/2116/commits/87d327d633b6726581bbc1e0cb42fa4ca0eed153

Reply:
>     * open a ticket to replace `From<[u8; 32]> for Nullifier` with `TryFrom<[u8; 32]> for Nullifier`, and assign it to Deirdre

Please open a follow-up ticket for this change.

Reply:
@dconnolly this is now https://github.com/ZcashFoundation/zebra/issues/2175